test/e2e/framework.go

```diff
@@ -525,25 +525,40 @@ func (data *TestData) collectExternalInfo() error {
 		}
 	}
 
-	subnets := strings.Split(testOptions.vlanSubnets, ",")
-	for _, subnet := range subnets {
-		if subnet == "" {
+	vlanSubnetsList := strings.Split(testOptions.vlanSubnets, ",")
+	for _, vlanSubnets := range vlanSubnetsList {
+		if vlanSubnets == "" {
 			continue
 		}
-		gatewayIP, _, err := net.ParseCIDR(subnet)
+		parts := strings.Split(vlanSubnets, "=")
+		if len(parts) != 2 {
+			return fmt.Errorf("invalid vlan subnet format: %s", vlanSubnets)
+		}
+		vlanIDStr := parts[0]
+		vlanID, err := strconv.Atoi(vlanIDStr)
 		if err != nil {
-			return fmt.Errorf("invalid vlan subnet %s: %w", subnet, err)
+			return fmt.Errorf("invalid vlan id %s: %w", vlanIDStr, err)
 		}
-		if gatewayIP.To4() != nil {
-			externalInfo.vlanSubnetIPv4 = subnet
-			externalInfo.vlanGatewayIPv4 = gatewayIP.String()
-		} else {
-			externalInfo.vlanSubnetIPv6 = subnet
-			externalInfo.vlanGatewayIPv6 = gatewayIP.String()
+		subnets := strings.Split(parts[1], ",")
+		for _, subnet := range subnets {
+			if subnet == "" {
+				continue
+			}
+
+			gatewayIP, _, err := net.ParseCIDR(subnet)
+			if err != nil {
+				return fmt.Errorf("invalid vlan subnet %s: %w", subnet, err)
+			}
+			if gatewayIP.To4() != nil {
+				externalInfo.vlanSubnetIPv4 = subnet
+				externalInfo.vlanGatewayIPv4 = gatewayIP.String()
+			} else {
+				externalInfo.vlanSubnetIPv6 = subnet
+				externalInfo.vlanGatewayIPv6 = gatewayIP.String()
+			}
 		}
+		externalInfo.vlanID = vlanID
```

Comment:
shouldn't the `flag.StringVar` declarations for the two options be updated/removed?